lab_3_ann_retriever/main.py

```diff
@@ -427,6 +574,37 @@ def build(self, vectors: list[Vector]) -> bool:
         In case of corrupt input arguments, False is returned.
         """
 
+        if not vectors or not all(isinstance(vec, tuple) and len(vec) > 0 for vec in vectors):
+            return False
+        dimensions = len(vectors[0])
+        another_vec = vectors[:]
+        nodes: list[tuple[list[Vector], int, Node | None, bool]] = [(vectors, 0, None, True)]
+        while nodes:
+            current_vectors, depth, parent, is_left = nodes.pop()
+            if not current_vectors:
+                continue
+            axis = depth % dimensions
+            try:
```

Comment:
can you do it without these try except cases?

Reply:
Done